src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -0,0 +1,265 @@
+import { getSASKey, LinkDuration } from '../../utils/azurestorage';
+import { parseResource } from '../../../azure-api-utils';
+import * as ast from 'azure-arm-storage';
+import { PeriscopeStorage, PeriscopeHTMLInterface } from '../models/storage';
+import * as amon from 'azure-arm-monitor';
+import * as htmlhandlers from "handlebars";
+import * as path from 'path';
+import * as fs from 'fs';
+import AksClusterTreeItem from '../../../tree/aksClusterTreeItem';
+const tmp = require('tmp');
+
+const {
+    BlobServiceClient,
+    StorageSharedKeyCredential
+} = require("@azure/storage-blob");
+
+export async function getClusterDiagnosticSettings(
+    cluster: AksClusterTreeItem
+): Promise<amon.MonitorManagementModels.DiagnosticSettingsCategoryResourceCollection | undefined> {
+    try {
+        // Get daignostic setting via diagnostic monitor
+        const diagnosticMonitor = new amon.MonitorManagementClient(cluster.root.credentials, cluster.root.subscriptionId);
+        const diagnosticSettings = await diagnosticMonitor.diagnosticSettingsOperations.list(cluster.id!);
+
+        return diagnosticSettings;
+    } catch (e) {
+        vscode.window.showErrorMessage(`Error fetching cluster diagnostic monitor: ${e}`);
+        return undefined;
+    }
+}
+
+export async function chooseStorageAccount(
+    diagnosticSettings: amon.MonitorManagementModels.DiagnosticSettingsResourceCollection | undefined,
+): Promise<string | undefined> {
+    /*
+        Check the diagnostic setting is 1 or more than 1:
+          1. For the scenario of 1 storage account in diagnostic settings - Pick the storageId resource and get SAS.
+          2. For the scenario for more than 1 then show VsCode quickPick to select and get SAS of selected.
+    */
+
+    if (diagnosticSettings && diagnosticSettings.value!.length > 1) {
+        const storageAccountNameToStorageIdArray: { id: string; label: string; }[] = [];
+
+        diagnosticSettings.value?.forEach((item) => {
+            if (item.storageAccountId) {
+                const { name } = parseResource(item.storageAccountId!);
+                if (!name) {
+                    vscode.window.showInformationMessage(`Storage Id is malformed: ${item.storageAccountId}`);
+                    return;
+                }
+                storageAccountNameToStorageIdArray.push({ id: item.storageAccountId, label: name });
+            }
+        });
+
+        // accounts is now an array of {id, name}
+        const accountQuickPicks = storageAccountNameToStorageIdArray;
+
+        // Create quick pick for more than 1 storage account scenario.
+        const selectedQuickPick = await vscode.window.showQuickPick(
+            accountQuickPicks,
+            {
+                placeHolder: "Select storage account for Periscope deployment:",
+                ignoreFocusOut: true,
+                onDidSelectItem: (item: string) => { return item.toString(); }
+            });
+
+        if (selectedQuickPick?.label) {
```

Comment:
Can't this just be `if (selectedQuickPick)`?  There shouldn't be any picks without labels.

Reply:
Then you can drop the `?` on the next line because the compiler will know that `selectedQuickPick` is non-null.